src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareFunction(a, b) {
+    if (String(a) > String(b)) {
+      return 1;
+    } else if (String(a) === String(b)) {
+      return 0;
+    } else {
+      return -1;
+    }
+  };
+
+  [].__proto__.sort2 = function (callback = compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (callback(this[j], this[i]) < 0) {
+          const temp = this[i];
+
+          this[i] = this[j];
+          this[j] = temp;
+        }
+      }
+    }
```

Comment:
you can simplify this code to use only one for a loop because now it iterates too many times (for every symbol it goes `this.length` times)